### PR TITLE
Add new Logstash chart

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juan@bitnami.com>

This PR adds a new Helm chart to the catalog: **Logstash**

## What is Logstash?

Logstash is an open source, server-side data processing pipeline that ingests data from a multitude of sources simultaneously, transforms it, and then sends it to your favorite "stash".

### Special Notes for Reviewers

Do not merge this PR until the Bitnami Docker Logstash Exporter image is released.

### CODE REVIEW

File: bitnami/logstash/README.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+| `fullnameOverride`                         | String to fully override logstash.fullname template with a string                                                    | `nil`                                                   |
+| `clusterDomain`                            | Default Kubernetes cluster domain                                                                                    | `cluster.local`                                         |
+| `configFileName`                           | Logstash configuration file name. It must match the name of the configuration file mounted as a configmap.           | `logstash.conf`                                         |
+| `enableMonitoringAPI`                      | Whether to enable the Logstash Monitoring API or not  Kubernetes cluster domain                                      | `true`                                                  |
+| `monitoringAPIPort`                        | Logstash Monitoring API Port                                                                                         | `9090`                                                  |
```
Comment (rafariossaa) [inline bitnami/logstash/README.md]: Shouldn't we keep the default port 9600 ?

File: bitnami/logstash/templates/configuration-cm.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  name: {{ include "logstash.fullname" . }}
+  labels: {{- include "logstash.labels" . | nindent 4 }}
+data:
+{{- if (.Files.Glob "files/conf/*") }}
+{{ (.Files.Glob "files/conf/*").AsConfig | indent 2 }}
+{{- else if (or .Values.input .Values.filter .Values.output) }}
+  logstash.conf: |-
+{{- if .Values.input }}
+    input {
+{{ tpl .Values.input . | indent 6 }}
```
Comment (javsalgar) [inline bitnami/logstash/templates/configuration-cm.yaml]: Change `.` to `$` for consistency

File: bitnami/logstash/templates/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    {{- range $key, $value := .Values.ingress.annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+spec:
+  rules:
+  {{- range .Values.ingress.hosts }}
+    - host: {{ .name }}
+      http:
+        paths:
+          - path: /
```
Comment (javsalgar) [inline bitnami/logstash/templates/ingress.yaml]: In other charts we allow `/` to be configurable

File: bitnami/logstash/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    Logstash Prometheus Metrics: {{ include "logstash.fullname" . }}-metrics.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }} (port {{ .Values.metrics.service.port }})
+    {{- end }}
+
+
+To access Logstash from outside the cluster execute the following commands:
+
+{{- if .Values.ingress.enabled }}
+{{- $ingressHost := first .Values.ingress.hosts }}
+
+  You should be able to access your new InfluxDB server(s) through:
```
Comment (javsalgar) [inline bitnami/logstash/templates/NOTES.txt]: InfluxDB -> Logstash

File: bitnami/logstash/values.yaml [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+    ## Service type
+    ##
+    type: ClusterIP
+    ## Logstash Prometheus port
+    ##
+    port: 9198
+    ## Specify the nodePort value for the LoadBalancer and NodePort service types.
+    ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport
+    ##
+    nodePort: ""
```
Comment (javsalgar) [inline bitnami/logstash/values.yaml]: You are not allowing nodePort in the normal service

Comment (juan131) [reply bitnami/logstash/values.yaml]: The ports defined in the "normal" services are totally customizable. Therefore, you can add the `nodePort` there

File: bitnami/logstash/templates/pdb.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  name: {{ include "logstash.fullname" . }}
+  labels: {{- include "logstash.labels" . | nindent 4 }}
+spec:
+  {{- if .Values.podDisruptionBudget.minAvailable }}
+  minAvailable: {{ .Values.podDisruptionBudget.minAvailable }}
+  {{- end }}
+  {{- if .Values.podDisruptionBudget.maxUnavailable }}
+  maxUnavailable: {{ .Values.podDisruptionBudget.maxUnavailable }}
+  {{- end }}
+  selector: {{ include "logstash.matchLabels" . | nindent 4 }}
```
Comment (javsalgar) [inline bitnami/logstash/templates/pdb.yaml]:
```
WARN - logstash/templates/pdb.yaml contains an invalid PodDisruptionBudget - app.kubernetes.io/instance: Additional property app.kubernetes.io/instance is not allowed
WARN - logstash/templates/pdb.yaml contains an invalid PodDisruptionBudget - app.kubernetes.io/name: Additional property app.kubernetes.io/name is not allowed
```

File: bitnami/logstash/templates/sts.yaml [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    {{- end }}
+  selector:
+    matchLabels: {{- include "logstash.matchLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels: {{- include "logstash.labels" . | nindent 8 }}
+      annotations:
+        checksum/configuration: {{ include (print $.Template.BasePath "/configuration-cm.yaml") . | sha256sum }}
+    spec:
+{{- include "logstash.imagePullSecrets" . | nindent 6 }}
```
Comment (javsalgar) [inline bitnami/logstash/templates/sts.yaml]: I would indent this so it is under spec

File: bitnami/logstash/templates/ingress.yaml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "logstash.fullname" . }}
+  labels: {{ include "logstash.labels" . | nindent 4 }}
+  annotations:
+    {{- if .Values.ingress.certManager }}
+    kubernetes.io/tls-acme: "true"
+    {{- end }}
+    {{- range $key, $value := .Values.ingress.annotations }}
```
Comment (javsalgar) [inline bitnami/logstash/templates/ingress.yaml]: This generates a warning in helm 

```
2019/10/17 16:31:28 Warning: Merging destination map for chart 'logstash'. The destination item 'annotations' is a table and ignoring the source 'annotations' as it has a non-table value of: <nil>
```

Example:

```
  annotations:
    kubernetes.io/ingress.class: nginx

```

Comment (javsalgar) [reply bitnami/logstash/templates/ingress.yaml]: Actually, it is not the range but setting the annotation value in ingress

File: bitnami/logstash/templates/sts.yaml (hunk 7 above)
Comment (juan131) [reply bitnami/logstash/templates/sts.yaml]: We can discus this in other moment. No other chart indent this field.

**Pull request merged**